### ISSUE TITLE
decoder-only sink mask update policy

### ISSUE THREAD

**Issue by ZJLi2013** [tomaarsen/attention_sinks#48]
hi, author,

just check the code about the mask update logic here:

```py
        # update decoder attention mask
        if "decoder_attention_mask" in model_kwargs:
            decoder_attention_mask = model_kwargs["decoder_attention_mask"]
            # assert decoder_attention_mask.size() == (bs, 1, q_len, kv_seqlen)
            model_kwargs["decoder_attention_mask"] = torch.cat(
                [decoder_attention_mask, decoder_attention_mask.new_ones((decoder_attention_mask.shape[0], 1))], dim=-1,)
```

it's a little conflict with my understand about casual(sink) mask, looks it gives mask in following:

```yml
        [
        [1, 0, 0, 0, 1],
        [1, 1, 0, 0, 1],
        [1, 1, 1, 0, 1],
        [1, 1, 1, 1, 1]
        ]
``` 

thanks for explain here
ZJ
